### PR TITLE
DT-1540, DT-1504, DT-1430, DT-1446, DT-1447, DT-1440, DT-1444, DT-1513, DT-1443 and DT-1496

### PR DESCRIPTION
**Pull request by HenrikHL**
This PR includes:
- [DT-1540](https://dcsa.atlassian.net/browse/DT-1540) - Modify the semantics for the `202` responses in both Booking and eBL APIs
- [DT-1504](https://github.com/dcsaorg/DCSA-OpenAPI/pull/320/commits/0dd29f70cfd3e8be74b53ce4ba60095b98ddb6e6) - Make `commoditySubreference` optional in SI
- [DT-1496](https://dcsa.atlassian.net/browse/DT-1496) - Remove `minItems=1` on optional `displayedAddress`
- [DT-1430](https://dcsa.atlassian.net/browse/DT-1430) - Extends `partyCode` maxLength from 100 --> 150
- [DT-1446](https://dcsa.atlassian.net/browse/DT-1446) - Modify the condition on `numberOfOriginalsXXX` description
- [DT-1447](https://dcsa.atlassian.net/browse/DT-1447) - Remove condition on `numberOfCopiesXXX` description
- [DT-1440](https://dcsa.atlassian.net/browse/DT-1440) - Update `sealSource` description and add condition
- `ShippingMarks` description updated on the `UtilizedTransportEquipment` object (was a copy/paste error)
- [DT-1444](https://dcsa.atlassian.net/browse/DT-1444) - Added `displayedShippedOnBoardReceivedForShipment` optional property to TD
- [DT-1513](https://dcsa.atlassian.net/browse/DT-1513) - Change `sendToPlatform` to a `IssueTo` party (`issueTo` has been added to `DocumentParties` structure as it is a Party instead of being on root level)
- [DT-1443](https://dcsa.atlassian.net/browse/DT-1443) - Fix wrong mentioning of `requestedChanges` should be `Feedback`

[DT-1540]: https://dcsa.atlassian.net/browse/DT-1540?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DT-1504]: https://dcsa.atlassian.net/browse/DT-1504?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DT-1496]: https://dcsa.atlassian.net/browse/DT-1496?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DT-1430]: https://dcsa.atlassian.net/browse/DT-1430?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DT-1446]: https://dcsa.atlassian.net/browse/DT-1446?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DT-1447]: https://dcsa.atlassian.net/browse/DT-1447?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DT-1440]: https://dcsa.atlassian.net/browse/DT-1440?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DT-1444]: https://dcsa.atlassian.net/browse/DT-1444?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DT-1513]: https://dcsa.atlassian.net/browse/DT-1513?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: bkg/v2/BKG_v2.0.0.yaml [hunk 1]
```diff
@@ -662,41 +730,74 @@ paths:
...
+
+            GET /v2/bookings/{bookingReference}?amendedContent=true
+
+        It is possible to `GET` the content of the `Amended Booking` via the example above until either:
+
+        - the provider requests for a new amendment (**UseCase 6: Request to amend confirmed Booking**) in which case the "old update" is no longer accessible
+        - the consumer submits a new update (**UseCase 7: Request amendment to confirmed Booking**) in which case the "new amendment" provided **replaces** the "old amendment".
+        - the provider re-confirms the `Booking` (**UseCase 5: Confirm booking request**) in which case the "old update" is no longer accessable.
+
+        ## Flow for a `202` (OK) response
```
Comment (palatsangeetha) [inline bkg/v2/BKG_v2.0.0.yaml:762]: 200 OK ? or 202 Accepted?

File: bkg/v2/BKG_v2.0.0.yaml [hunk 2]
```diff
@@ -662,41 +730,74 @@ paths:
...
+
+        For `POST` `Booking Request` responding with `202` (Accepted) the process ends here. The Booking Request:
+          - is now accepted by the provider system
+          - the status of the `Booking Request` is unchanged.
+          - a 202 (Accepted) response is sent with an empty payload
+          - awaits further processing by the provider
+        
+        The provider will now start asynchronous processing. Once processed, the state will change to one of the following values depending on the UseCase for calling the `PUT` endPoint:
+          - `UPDATE RECEIVED` in case this endPoint has been used to:
+            - send an update to a newly created Booking (preconition: `bookingStatus='RECEIVED'`)
```
Comment (palatsangeetha) [inline bkg/v2/BKG_v2.0.0.yaml:780]: precondition (typo)

File: bkg/v2/BKG_v2.0.0.yaml [hunk 3]
```diff
@@ -662,41 +730,74 @@ paths:
...
+            - send an update to a Booking because of a request from provider via **UseCase 2 - Request to update Booking request** (preconition: `bookingStatus='PENDING UPDATE'`)
```
Comment (palatsangeetha) [inline bkg/v2/BKG_v2.0.0.yaml:781]: same here

File: bkg/v2/BKG_v2.0.0.yaml [hunk 4]
```diff
@@ -662,41 +730,74 @@ paths:
...
+            - send an additional update to a Booking that has already received an update  (preconition: `bookingStatus='UPDATE RECEIVED'`)
+          - `bookingStatus` will stay as `PENDING AMENDMENT` but `amendedBookingStatus` will change to `AMENDMENT RECEVIED` in case this endPoint is being used in response to a request from provider via **UseCase 6 - Request to amend Confirmed Booking** (preconition: `bookingStatus='PENDING AMENDMENT'`)
```
Comment (palatsangeetha) [inline bkg/v2/BKG_v2.0.0.yaml:783]: same as above in all the usecases

File: bkg/v2/BKG_v2.0.0.yaml [hunk 5]
```diff
@@ -1807,42 +1935,53 @@ paths:
...
-                  value:
-                    carrierBookingRequestReference: cbrr-123
-                    carrierBookingReference: cbr-987
-                    bookingStatus: CONFIRMED
-                    amendedBookingStatus: AMENDMENT CANCELLED
-                amendmentCancelled2Example:
+                    The consumer has requested that the amendment to a confirmed `Booking` is to be cancelled. The cancellation has been accepted and is now awaiting further processing by the provider
+        '400':
+          description: |
+            In case the Cancel payload does not schema validate a `400` (Bad Request) is returned
```
Comment (palatsangeetha) [inline bkg/v2/BKG_v2.0.0.yaml:1951]: Do we need to Capitalize 'C' in Cancel here?

File: ebl/v3/EBL_v3.0.0.yaml [hunk 6]
```diff
@@ -88,33 +87,61 @@ paths:
...
+        For `POST` `Shipping Instructions` responding with `201` (Created) the process ends here. The `Shipping Instructions`:
+          - is now stored in the provider system
+          - has status `RECEIVED` and the status can be queried (a subsequent `GET` request will retrieve the payload)
+          - a `201` (Created) response is sent to the consumer with a payload containing the `shippingInstructionsReference` and `shippingInstructionsStatus='RECEIVED'`
+          - '_relevant_' subscribers have been notified.
+          - awaits further processing by the provider
+
+        The provider will now start asynchronous processing. Once processed, the new status of the `Shipping Instructions` will be communicated via a [Shipping Instructions Notification](#/ShippingInstructionsNotification). In case the consumer does not subscribe to notifications it is necessary for the consumer to poll on the
+
+            GET /v3/shipping-instructions/{documentReference}
```
Comment (palatsangeetha) [inline ebl/v3/EBL_v3.0.0.yaml:108]: Do we need to highlight the endpoints?

File: ebl/v3/EBL_v3.0.0.yaml [hunk 7]
```diff
@@ -193,23 +253,35 @@ paths:
                     shippingInstructionsStatus: RECEIVED
         '202':
           description: |
-            The `Shipping Instructions` has been accepted by the provider. The status of the `Shipping Instructions` is now `RECEIVED` - it is not possible to call the `GET` endPoint until the `Shipping Instructions` is further processed in provider system. The consumer is now awaiting provider to process the `Shipping Instructions` asynchronously.
+            The `Shipping Instructions` has been accepted by the provider. The `Shipping Instructions` does not yet have a `shippingInstructionsStatus` - it is not possible to call the `GET` endPoint until the `Shipping Instructions` is further processed in provider system. The consumer is now awaiting provider to process the `Shipping Instructions` asynchronously..
```
Comment (palatsangeetha) [inline ebl/v3/EBL_v3.0.0.yaml:256]: Extra '.' at the end.

File: ebl/v3/EBL_v3.0.0.yaml [hunk 8]
```diff
@@ -359,46 +431,71 @@ paths:
...
+
+        ## Flow for a `202` (Accepted) response
+        The following occurs when a provider receives an **update** to a `Shipping Instructions`
+        1. The payload (`Updated Shipping Instructions`) is schema-validated. In case the payload **is invalid** a `400` (Bad Request) is returned.
+
+           **The process stops here!**
+        2. The payload is schema-valid which means:
+            - all required properties are provided
+            - all values provided have correct data type.
+        3. An empty response is returned and the consumer now awaits futher processing by the provider.
```
Comment (palatsangeetha) [inline ebl/v3/EBL_v3.0.0.yaml:476]: Typo - further

File: bkg/v2/BKG_v2.0.0.yaml [hunk 9]
```diff
@@ -83,34 +81,61 @@ paths:
...
+
+        For `POST` `Booking Request` responding with `202` (Accepted) the process ends here. The `Booking Request`:
+          - is now accepted by the provider system
+          - the `Booking Request` does not yet have any status and cannot be queried (no `GET` request is possible until the `Booking Request` is further processed in the provider system)
+          - a `202` (Accepted) response is sent to the consumer with a payload **only** containing the `carrierBookingRequestReference`
+          - awaits further processing by the provider
+
+        The provider will now start asynchronous processing. Once processed, the status `RECEIVED` of the `Booking Request` will be communicated via a [Booking Notification](#/BookingNotification). In case the consumer does not subscribe to notifications it is necessary for the consumer to poll on the
+
+            GET /v2/bookings/{bookingReference}
```
[the remark this answers is not in the document]
Comment (HenrikHL) [reply bkg/v2/BKG_v2.0.0.yaml:134]: The highlight should be there when the description is indented 4 spaces (then it is marked as `code` )

File: bkg/v2/BKG_v2.0.0.yaml (hunk 1 above)
Comment (HenrikHL) [reply bkg/v2/BKG_v2.0.0.yaml:762]: Well spotted!!!

File: bkg/v2/BKG_v2.0.0.yaml (hunk 3 above)
Comment (HenrikHL) [reply bkg/v2/BKG_v2.0.0.yaml:781]: fixed

File: bkg/v2/BKG_v2.0.0.yaml (hunk 2 above)
Comment (HenrikHL) [reply bkg/v2/BKG_v2.0.0.yaml:780]: fixed

File: bkg/v2/BKG_v2.0.0.yaml (hunk 4 above)
Comment (HenrikHL) [reply bkg/v2/BKG_v2.0.0.yaml:783]: fixed

File: bkg/v2/BKG_v2.0.0.yaml (hunk 5 above)
Comment (HenrikHL) [reply bkg/v2/BKG_v2.0.0.yaml:1951]: changed to small :-)

File: ebl/v3/EBL_v3.0.0.yaml (hunk 6 above)
Comment (HenrikHL) [reply ebl/v3/EBL_v3.0.0.yaml:108]: Indenting with 4 spaces should mark it as `code`...

File: ebl/v3/EBL_v3.0.0.yaml (hunk 7 above)
Comment (HenrikHL) [reply ebl/v3/EBL_v3.0.0.yaml:256]: Well spotted

File: ebl/v3/EBL_v3.0.0.yaml (hunk 8 above)
Comment (HenrikHL) [reply ebl/v3/EBL_v3.0.0.yaml:476]: fixed

File: ebl/v3/issuance/EBL_ISS_v3.0.0.yaml [hunk 10]
```diff
@@ -570,27 +570,31 @@ components:
...
           minimum: 0
-          description: The requested number of copies of the Transport document to be issued by the carrier including charges. Only applicable for physical (paper) documents
+          description: The requested number of copies of the `Transport Document` to be issued by the carrier including charges.
           example: 2
         numberOfCopiesWithoutCharges:
           type: integer
           format: int32
           minimum: 0
-          description: The requested number of copies of the Transport document to be issued by the carrier **NOT** including charges. Only applicable for physical (paper) documents
+          description: The requested number of copies of the`Transport Document` to be issued by the carrier **NOT** including charges.
```
Comment (jkosternl) [inline ebl/v3/issuance/EBL_ISS_v3.0.0.yaml:579]:
```suggestion
          description: The requested number of copies of the `Transport Document` to be issued by the carrier **NOT** including charges.
```

Comment (HenrikHL) [reply ebl/v3/issuance/EBL_ISS_v3.0.0.yaml:579]: Thanks :-)

**Review by palatsangeetha — APPROVED** [review]
Changes looks good.

**Pull request merged**